independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorInfo.java

```diff
@@ -52,42 +53,78 @@
+        DotName superTypeName = aClass.superName();
+        if (superTypeName != null && !DotNames.OBJECT.equals(superTypeName)) {
+            ClassInfo superClassInfo = getClassByName(beanDeployment.getBeanArchiveIndex(), superTypeName);
+            if (superClassInfo == null) {
+                throw new DefinitionException(
+                        "The interception super type " + superTypeName + " was not found in the index");
+            }
+            processInheritedMethods(superClassInfo, beanDeployment);
+        }
+
+        for (MethodInfo method : aClass.methods()) {
+            if (Modifier.isStatic(method.flags())) {
+                continue;
+            }
+            if (method.hasAnnotation(DotNames.AROUND_INVOKE)) {
+                //Map can not been used because order must be respected.
+                //remove overridden methods
+                for (MethodInfo info : aroundInvokes) {
+                    if (info.name().equals(method.name())) {
+                        aroundInvokes.remove(info);
+                        break;
+                    }
                 }
-                if (method.hasAnnotation(DotNames.AROUND_CONSTRUCT)) {
-                    aroundConstructs.add(validateSignature(method));
+                aroundInvokes.add(validateSignature(method));
+            }
+            if (method.hasAnnotation(DotNames.AROUND_CONSTRUCT)) {
+                //Map can not been used because order must be respected.
+                //remove overridden methods
+                for (MethodInfo info : aroundConstructs) {
+                    if (info.name().equals(method.name())) {
+                        aroundConstructs.remove(info);
+                        break;
+                    }
                 }
-                if (method.hasAnnotation(DotNames.POST_CONSTRUCT)) {
-                    postConstructs.add(validateSignature(method));
+                aroundConstructs.add(validateSignature(method));
+            }
+            if (method.hasAnnotation(DotNames.POST_CONSTRUCT)) {
+                //Map can not been used because order must be respected.
+                //remove overridden methods
+                for (MethodInfo info : postConstructs) {
+                    if (info.name().equals(method.name())) {
+                        postConstructs.remove(info);
+                        break;
+                    }
                 }
-                if (method.hasAnnotation(DotNames.PRE_DESTROY)) {
-                    preDestroys.add(validateSignature(method));
+                postConstructs.add(validateSignature(method));
+            }
+            if (method.hasAnnotation(DotNames.PRE_DESTROY)) {
+                //Map can not been used because order must be respected.
+                //remove overridden  methods
+                for (MethodInfo info : preDestroys) {
+                    if (info.name().equals(method.name())) {
```

Comment:
A subclass's method can't generally be considered an override of a superclass's method just because it has the same name. In case of interceptors, it's much simpler, because the interceptor methods have a well known signature, so in theory, checking the name should be enough. Except I think we have an extra feature (the invocation context parameter of the interceptor method may also be `ArcInvocationContext`) that makes this wrong.

Am I right? Genuine question.

Actually I'm probably wrong, because that would be an invalid override. So probably just ignore me here :-)

Reply:
hmm for me signature was the same... my bad. I will think to it.

Reply:
I was just thinking out loud here. I think it's worth highlighting what you're doing here -- even if it's correct.